From 012ba2f861c62eb0705959095d82a5ffd673124d Mon Sep 17 00:00:00 2001
From: Jiaqing Zhao <jiaqing.zhao@intel.com>
Date: Fri, 12 Jan 2024 07:18:59 +0000
Subject: [PATCH 41/54] Put diskbus to BDF convertion in a separate function

SBL diskbus to BDF convertion may be used multiple times in some
projects, it's better to be put in a separate function.

Tracked-On: OAM-114775
Signed-off-by: Jiaqing Zhao <jiaqing.zhao@intel.com>
---
 libkernelflinger/android.c | 33 ++++++++++++++++++---------------
 1 file changed, 18 insertions(+), 15 deletions(-)

diff --git a/libkernelflinger/android.c b/libkernelflinger/android.c
index 81b6be3..5710b6b 100644
--- a/libkernelflinger/android.c
+++ b/libkernelflinger/android.c
@@ -1249,6 +1249,18 @@ UINT32 __attribute__((weak)) get_bootdev_diskbus()
 	return 0;
 
 }
+
+static INT32 diskbus_to_bdf(UINT32 diskbus)
+{
+        UINT32 storage_bus_num;
+
+        storage_bus_num = diskbus >> 16;
+        if (storage_bus_num == 0) {
+                return (diskbus >> 8);
+        }
+
+        return bridge_diskbus(storage_bus_num);
+}
 #endif
 
 /* when we call setup_command_line in EFI, parameter is EFI_GUID *swap_guid.
@@ -1386,22 +1398,13 @@ static EFI_STATUS setup_command_line(
                 CHAR16 *diskbus = NULL;
 #ifdef AUTO_DISKBUS
 #ifdef USE_SBL
-		INT32 bdf;
-		UINT32 disk_bus_num, storage_bus_num;
-
-		disk_bus_num = get_bootdev_diskbus();
-		debug(L"diskbus = %x", disk_bus_num);
+                INT32 bdf;
 
-		storage_bus_num = disk_bus_num >> 16;
-                if (storage_bus_num == 0) {
-                        bdf = (disk_bus_num >> 8);
-                } else {
-                        bdf = bridge_diskbus(storage_bus_num);
-                        if (bdf < 0) {
-                                error(L"No pci bridge found");
-                                ret = EFI_INVALID_PARAMETER;
-                                goto out;
-                        }
+                bdf = diskbus_to_bdf(get_bootdev_diskbus());
+                if (bdf < 0) {
+                        error(L"No pci bridge found");
+                        ret = EFI_INVALID_PARAMETER;
+                        goto out;
                 }
 
                 diskbus = PoolPrint(L"%02x.%x", (bdf >> 3) & 0x1f, bdf & 0x7);
-- 
2.25.1

